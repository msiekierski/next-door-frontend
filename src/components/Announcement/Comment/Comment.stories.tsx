import React from "react";
import Comment from "./Comment";

export default {
  title: "Components/Post/Comment",
  component: Comment,
  argTypes: {
    author: {
      control: "text",
    },
    date: {
      control: "date",
    },
    description: {
      control: "text",
    },
  },
};

export const comment = (args: any) => <Comment {...args} />;
comment.args = {
  author: "Micha≈Ç Sznajder",
  date: 0,
  description: "i disagree",
};
