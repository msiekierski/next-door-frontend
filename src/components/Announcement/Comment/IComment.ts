interface IComment {
  idAccount?: number;
  idAnnouncement?: number;
  creationDate: string;
  description: string;
  name: string;
  surname: string;
}

export default IComment;
