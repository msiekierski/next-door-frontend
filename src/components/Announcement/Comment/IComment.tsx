interface IComment {
  id: number;
  author: string;
  date: number;
  description: string;
}

export default IComment;
