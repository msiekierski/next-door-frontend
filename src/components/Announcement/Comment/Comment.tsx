import React, { FunctionComponent } from "react";
import { Card } from "react-bootstrap";
import IComment from "./IComment";

type Props = IComment;

const Comment: FunctionComponent<Props> = ({ description, creationDate }) => {
  return (
    <Card>
      <Card.Body>
        <Card.Title className={`d-flex justify-content-between`}>
          <Card.Subtitle className={`text-muted`}></Card.Subtitle>
          <Card.Subtitle className={`text-muted text-right`}>{creationDate}</Card.Subtitle>
        </Card.Title>
        <Card.Text>{description}</Card.Text>
      </Card.Body>
    </Card>
  );
};

export default Comment;
