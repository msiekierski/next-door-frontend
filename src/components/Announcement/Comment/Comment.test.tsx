import React from "react";
import renderer from "react-test-renderer";
import Comment from "./Comment";

it("renders correctly", () => {
  const tree = renderer.create(<Comment creationDate={"test date"} description={"test desc"} />);
  expect(tree).toMatchSnapshot();
});
