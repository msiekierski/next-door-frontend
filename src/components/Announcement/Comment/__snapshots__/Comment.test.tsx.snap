// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="d-flex justify-content-between card-title h5"
    >
      <div
        className="text-muted card-subtitle h6"
      >
        Author
      </div>
      <div
        className="text-muted text-right card-subtitle h6"
      >
        Invalid Date
        <br />
        Invalid Date
      </div>
    </div>
    <p
      className="card-text"
    >
      test desc
    </p>
  </div>
</div>
`;
